Signal/src/ViewControllers/NewGroupViewController.m

```diff
@@ -66,243 +79,353 @@ - (instancetype)initWithCoder:(NSCoder *)aDecoder
+
+                    return cell;
+                }
+                            customRowHeight:[ContactTableViewCell rowHeight]
+                            actionBlock:^{
+                                SignalAccount *signalAccount = [helper signalAccountForRecipientId:recipientId];
+                                if (signalAccount) {
+                                    [weakSelf.groupViewHelper
+                                        showRemoveFromGroupAlertForSignalAccount:signalAccount
+                                                              fromViewController:weakSelf
+                                                                 contactsManager:helper.contactsManager
+                                                                    successBlock:^{
+                                                                        [weakSelf removeSignalAccount:signalAccount];
+                                                                    }];
+                                } else {
+                                    [weakSelf.groupViewHelper
+                                        showRemoveFromGroupAlertForRecipientId:recipientId
+                                                            fromViewController:weakSelf
+                                                               contactsManager:helper.contactsManager
+                                                                  successBlock:^{
+                                                                      [weakSelf removeRecipientId:recipientId];
+                                                                  }];
+                                }
+                            }]];
+        }
+        [contents addSection:nonContactsSection];
     }
 
-    return NO;
-}
+    // Contacts
 
-- (BOOL)isContactBlocked:(Contact *)contact
-{
-    if (contact.parsedPhoneNumbers.count < 1) {
-        // Hide contacts without any valid phone numbers.
-        return NO;
-    }
+    OWSTableSection *signalAccountSection = [OWSTableSection new];
+    signalAccountSection.headerTitle = NSLocalizedString(
+        @"EDIT_GROUP_CONTACTS_SECTION_TITLE", @"a title for the contacts section of the 'new/update group' view.");
+    if (signalAccounts.count > 0) {
+
+        if (nonContactMemberRecipientIds.count < 1) {
+            // We always want to offer a way to add non-contacts.
+            [signalAccountSection addItem:[self createAddNonContactItem]];
+        }
 
-    for (PhoneNumber *phoneNumber in contact.parsedPhoneNumbers) {
-        if ([_blockedPhoneNumbers containsObject:phoneNumber.toE164]) {
-            return YES;
+        for (SignalAccount *signalAccount in signalAccounts) {
+            [signalAccountSection
+                addItem:[OWSTableItem itemWithCustomCellBlock:^{
+                    NewGroupViewController *strongSelf = weakSelf;
+                    if (!strongSelf) {
+                        return (ContactTableViewCell *)nil;
```

Comment:
Instead I'm just going to remove the strongSelf checks since they should never fail.